### PR TITLE
adds test_mirror_suite for testing mirroring on multiple nodes

### PR DESCRIPTION
**Pull request by zippy**
## PR summary

Creates a test suite for spinning up n of nodes, and then sees if when one node authors an entry, that all other nodes get the entry gossiped

## Review checklist 
- [ ] The story has unit or integration tests
- [ ] No new bugs, and any tech-debt is identified and justified
- [ ] There is enough API documentation (how to use)
- [ ] There is enough code documentation (how the code works)

### CODE REVIEW

File: crates/lib3h/tests/integration_test.rs [hunk 1]
```diff
@@ -266,6 +266,44 @@ fn launch_three_memory_nodes_test(test_fn: ThreeNodesTestFn, can_setup: bool) ->
...
+        launch_mirror_test(*test_fn, *can_setup).unwrap();
+    }
+}
+
+// Do general test with config
+fn launch_mirror_test(test_fn: MultiNodeTestFn, can_setup: bool) -> Result<(), ()> {
+    let test_fn_ptr = test_fn as *mut std::os::raw::c_void;
+    println!("");
+    print_test_name("IN-MEMORY MIRROR TEST: ", test_fn_ptr);
+    println!("==========================");
```
Comment (struktured) [inline crates/lib3h/tests/integration_test.rs]: Change printlns to debug/trace?

File: crates/lib3h/tests/integration_test.rs [hunk 2]
```diff
@@ -266,6 +266,44 @@ fn launch_three_memory_nodes_test(test_fn: ThreeNodesTestFn, can_setup: bool) ->
...
+    if can_setup {
+        setup_mirror_nodes(&mut nodes);
+    }
+
+    // Execute test
+    test_fn(&mut nodes);
+
+    // Wrap-up test
+    println!("==========================");
+    print_test_name("IN-MEMORY MIRROR TEST: ", test_fn_ptr);
```
Comment (struktured) [inline crates/lib3h/tests/integration_test.rs]: Remove or change to trace/debug?

File: crates/lib3h/tests/test_suites/mirror.rs [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+
+#[allow(dead_code)]
+pub fn setup_mirror_nodes(nodes: &mut Vec<NodeMock>) {
+    // Wait for nodes to auto-connect with Discovery
+    assert!(nodes.len() > 1);
+    let mut node0 = nodes.remove(0);
+    let mut node1 = nodes.remove(0);
+    let expected = "Connected\\(ConnectedData \\{ request_id: \"[\\w\\d_~]+\", uri: Lib3hUri\\(\".*\"\\) \\}\\)";
+    let _results = assert_msg_matches!(node1, expected);
+    wait_engine_wrapper_until_no_work!(node0);
```
Comment (struktured) [inline crates/lib3h/tests/test_suites/mirror.rs]: Would be nice to have this wait on some event instead but not sure what it has to process here..

File: crates/lib3h/tests/test_suites/mirror.rs [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+        println!("yes!");
+    }
+}
+
+fn process_nodes(nodes: &mut Vec<NodeMock>) {
+    for _i in 0..50 {
+        process_nodes_inner(nodes);
+    }
+}
+fn process_nodes_inner(nodes: &mut Vec<NodeMock>) {
```
Comment (struktured) [inline crates/lib3h/tests/test_suites/mirror.rs]: I would like to tackle baking this into the processor harness eventually but it's fine for this PR. Nice trick using a private function to fix scoping issues.

File: crates/lib3h/tests/test_suites/mirror.rs [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+// Tests
+//--------------------------------------------------------------------------------------------------
+
+/// Empty function that triggers the test suite
+#[allow(dead_code)]
+fn test_setup_only(_nodes: &mut Vec<NodeMock>) {
+    // n/a
+}
+
+fn test_mirror_from_center(nodes: &mut Vec<NodeMock>) {
```
Comment (struktured) [inline crates/lib3h/tests/test_suites/mirror.rs]: I don't know what `from_center` means here. Maybe a comment?

File: crates/lib3h/src/gateway/gateway_transport.rs [hunk 6]
```diff
@@ -18,6 +18,14 @@ use serde::{Deserialize, Serialize};
 impl P2pGateway {
     /// Handle IncomingConnection event from child transport
     fn handle_incoming_connection(&mut self, span: Span, uri: Lib3hUri) -> TransportResult<()> {
+        // TODO: This is prbably wrong in that a different level of URI should be being bubbled up.
```
Comment (struktured) [inline crates/lib3h/src/gateway/gateway_transport.rs]: Is there follow up work documented for this?

Comment (zippy) [reply crates/lib3h/src/gateway/gateway_transport.rs]: done

File: crates/lib3h/tests/integration_test.rs (hunk 1 above)
Comment (zippy) [reply crates/lib3h/tests/integration_test.rs]: there are a bunch more of these and it's just the tests so I'm ok leaving this for now.

**Pull request merged**